src/averageAges.js

```diff
@@ -56,6 +90,33 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const filterWomen = people.filter((person) => person.sex === 'f');
+
+  const mothers = [];
+  const ageDiffChildMother = [];
+  const ageDiffSonMother = [];
+
+  for (const key of filterWomen) {
+    people.some(child => {
+      if (key.name === child.mother) {
+        mothers.push(key);
+
+        if (onlyWithSon && child.sex === 'm') {
+          ageDiffSonMother.push(child.born - key.born);
+        }
+
+        ageDiffChildMother.push(child.born - key.born);
+      }
+    });
+  };
+
+  function checkWhoToCompare(compare) {
+    return compare.reduce((a, b) => a + b, 0) / compare.length;
+  }
```

Comment:
This function could be moved out as well and be used in calculateAverageAgeDiff and getAverageAge functions
```suggestion
  function getAverageValue(numbers) {
    return numbers.reduce((prevNumber, nextNumber) => prevNumber + nextNumber, 0) / numbers.length;
  }
```